torch/_inductor/util_autotuning_log_parser.py

```diff
@@ -0,0 +1,244 @@
+                    "Bias_shape",
+                ],
+                dropna=False,
+            ).benchmark_result.idxmin()
+        ]
+
+    @staticmethod
+    def strhash(s):
+        return hashlib.md5(s.encode()).hexdigest()
+
+    @staticmethod
+    def dim_info(size, stride):
+        if stride == 0:
+            return f"*{size}"
+        if stride == 1:
+            return f"!{size}"
+        return str(size)
+
+    @staticmethod
+    def parse_dtype(layout_str):
+        match: Optional[re.Match[str]] = re.search(
+            r"Layout\('cuda', torch\.([^, ]+),", layout_str
+        )
+        assert match is not None, f"Failed to parse dtype: {layout_str}"
+        return match.group(1)
+
+    @classmethod
+    def parse_record(cls, rec: Dict[str, Any]):
+        result = dict()
+        result["backend"] = rec["backend"]
+        result["name"] = rec.get("name", "ATen")
+        inputs = json.dumps(rec["input_nodes"])
+        result["problem_hash"] = cls.strhash(inputs)
+        result["kernel_schedule"] = rec.get("kernel_schedule", "")
+        result["tile_shape"] = rec.get("tile_shape", "[]")
+        result["benchmark_result"] = rec["benchmark_result"]
+        result["device"] = rec.get("device", "unknown")
+        result["cuda_device_name"] = rec["cuda_device_name"]
+        # result['keys'] = ",".join(rec.keys())
+        for name, tinfo in cls.parse_inputs(rec["input_nodes"], rec["backend"]).items():
+            if isinstance(tinfo, dict):
+                for key, value in tinfo.items():
+                    result[f"{name}_{key}"] = value
+            elif tinfo is not None:
+                result[name] = tinfo
+        return result
+
+    @staticmethod
+    def load_raw_records(path_or_fd: Union[str, io.TextIOWrapper]):
+        if isinstance(path_or_fd, str):
+            with open(path_or_fd, encoding="utf-8") as fh:
+                for line in fh:
+                    yield json.loads(line)
+        else:
+            if hasattr(path_or_fd, "seek"):
+                path_or_fd.seek(0)
+            for line in path_or_fd:
+                yield json.loads(line)
+
+    @classmethod
+    def load_records(
+        cls,
+        path_or_fd: Union[str, io.TextIOWrapper, List[Union[str, io.TextIOWrapper]]],
+    ):
+        if isinstance(path_or_fd, list):
+            for pf in path_or_fd:
+                yield from cls.load_records(pf)
+        else:
+            for record in cls.load_raw_records(path_or_fd):
+                yield cls.parse_record(record)
+
+    @staticmethod
+    def load_raw_grepped_records(path, pattern):
+        with open(path, encoding="utf-8") as fh:
+            for line in fh:
+                if pattern in line:
+                    yield json.loads(line)
+
+    @classmethod
+    def parse_layout(cls, layout_str):
+        match: Optional[re.Match[str]] = re.search(
+            r"size=\[([0-9]+), ([0-9]+)(, ([0-9]+))?\], stride=\[([0-9]+), ([0-9]+)(, ([0-9]+))?\]",
```

Comment:
If we leave the symbolic names (s0) in the size, this regexp will fail for dynamic shapes.

Reply:
True, see above.